src/main/java/dev/waterdog/waterdogpe/event/defaults/ServerTransferEvent.java

```diff
@@ -16,30 +16,38 @@
 package dev.waterdog.waterdogpe.event.defaults;
 
 import dev.waterdog.waterdogpe.network.serverinfo.ServerInfo;
-import dev.waterdog.waterdogpe.network.session.DownstreamClient;
 import dev.waterdog.waterdogpe.player.ProxiedPlayer;
 
 /**
  * Signalizing that player is being transferred to a new server.
  * This even is not cancellable. Use PreTransferEvent to cancel transfer.
  */
-public class PlayerTransferEvent extends PlayerEvent {
+public class ServerTransferEvent extends PlayerEvent {
 
-    private final ServerInfo oldServer;
-    private final DownstreamClient serverConnection;
+    private final ServerInfo sourceServer;
+    private final ServerInfo targetServer;
+    private boolean allowTransferScreen = true;
 
-    public PlayerTransferEvent(ProxiedPlayer player, ServerInfo oldServer, DownstreamClient serverConnection) {
+    public ServerTransferEvent(ProxiedPlayer player, ServerInfo sourceServer, ServerInfo targetServer) {
         super(player);
-        this.oldServer = oldServer;
-        this.serverConnection = serverConnection;
+        this.sourceServer = sourceServer;
+        this.targetServer = targetServer;
     }
 
-    public ServerInfo getOldServer() {
-        return this.oldServer;
+    public ServerInfo getSourceServer() {
+        return this.sourceServer;
     }
 
-    public DownstreamClient getServerConnection() {
-        return this.serverConnection;
+    public ServerInfo getTargetServer() {
+        return this.targetServer;
+    }
+
+    public void setAllowTransferScreen(boolean allowTransferScreen) {
+        this.allowTransferScreen = allowTransferScreen;
+    }
+
+    public boolean allowTransferScreen() {
```

Comment:
I think we should be consistent with getter method names. Since we use getSomething everywhere else, I suggest renaming this to getTransferScreenAllowed() or isTransferScreenAllowed()